runtime/src/common/sgx/avr.rs

```diff
@@ -314,44 +287,59 @@ pub fn verify(avr: &AVR) -> Fallible<AuthenticatedAVR> {
         }
     };
 
-    let quote_body = avr_body.isv_enclave_quote_body()?;
-    let quote_body = match base64::decode(&quote_body) {
-        Ok(quote_body) => quote_body,
-        _ => return Err(AVRError::MalformedQuote.into()),
-    };
-    let quote_body = match QuoteBody::decode(&quote_body) {
-        Ok(quote_body) => quote_body,
-        _ => return Err(AVRError::MalformedQuote.into()),
-    };
-
     #[cfg(target_env = "sgx")]
     {
+        let quote_body = avr_body.isv_enclave_quote_body()?;
+        let quote_body = match base64::decode(&quote_body) {
+            Ok(quote_body) => quote_body,
+            _ => return Err(AVRError::MalformedQuote.into()),
+        };
+        let quote_body = match QuoteBody::decode(&quote_body) {
+            Ok(quote_body) => quote_body,
+            _ => return Err(AVRError::MalformedQuote.into()),
+        };
+
         if option_env!("OASIS_UNSAFE_ALLOW_DEBUG_ENCLAVES").is_none() {
-            let af = match AttributesFlags::from_bits(u64::from_le_bytes(
-                quote_body.report_body.attributes[0..8].try_into()?,
-            )) {
-                Some(af) => af,
-                None => return Err(AVRError::MalformedReportAttributes.into()),
-            };
-
-            if af.contains(AttributesFlags::DEBUG) {
+            if quote_body
+                .report_body
+                .attributes
+                .flags
+                .contains(AttributesFlags::DEBUG)
+            {
                 return Err(AVRError::DebugEnclave.into());
             }
         }
+
+        // Force-ratchet the clock forward, to at least the time in the AVR.
+        update_insecure_posix_time(timestamp);
+
+        Ok(AuthenticatedAVR {
+            report_data: quote_body.report_body.reportdata.to_vec(),
+            identity: EnclaveIdentity {
+                mr_enclave: MrEnclave::from(quote_body.report_body.mrenclave.to_vec()),
+                mr_signer: MrSigner::from(quote_body.report_body.mrsigner.to_vec()),
+            },
+            timestamp,
+            nonce: nonce.to_string(),
+        })
     }
 
-    // Force-ratchet the clock forward, to at least the time in the AVR.
-    update_insecure_posix_time(timestamp);
-
-    Ok(AuthenticatedAVR {
-        report_data: quote_body.report_body.report_data,
-        identity: EnclaveIdentity {
-            mr_enclave: quote_body.report_body.mr_enclave,
-            mr_signer: quote_body.report_body.mr_signer,
-        },
-        timestamp,
-        nonce: nonce.to_string(),
-    })
+    #[cfg(not(target_env = "sgx"))]
+    {
+        // Force-ratchet the clock forward, to at least the time in the AVR.
+        update_insecure_posix_time(timestamp);
+
+        // Return dummy AVR.
+        Ok(AuthenticatedAVR {
+            report_data: vec![],
+            identity: EnclaveIdentity {
+                mr_enclave: MrEnclave::from(vec![]),
+                mr_signer: MrSigner::from(vec![]),
```

Comment:
was this desired? that we not do the usual checks and return a dummy AVR? I thought we use this in the gateway

Reply:
Yeah, this was a leftover of `Report` cleanup. SGX/non-SGX case can now be joined.